NewsApp/View/CustomTableViewCell.swift

```diff
@@ -12,25 +12,68 @@ class CustomTableViewCell: UITableViewCell {
     @IBOutlet weak var titleLabel: UILabel!
     @IBOutlet weak var authorLabel: UILabel!
     @IBOutlet weak var checkmarkImage: UIImageView!
+    @IBOutlet weak var favoritesButton: UIButton!
+    
+    var index: Int?
+    var viewModel: ArticleViewModel?
+    var coreDataService = CoreDataManager.shared
+    
+    var isFavorite: Bool = false {
+        didSet {
+            let favoriteImage = isFavorite ? UIImage(systemName: Constants.favoriteImage) : UIImage(systemName: Constants.unfavoriteImage)
+            favoritesButton.setImage(favoriteImage, for: .normal)
+        }
+    }
     
     override func awakeFromNib() {
         super.awakeFromNib()
-        checkmarkImage.image = UIImage(systemName: Constants.checkedImage)
+        checkmarkImage.image = UIImage(systemName: Constants.uncheckedImage)
         titleLabel.font = UIFont.boldSystemFont(ofSize: 15.0)
         authorLabel.font = UIFont.systemFont(ofSize: 15.0)
     }
     
-    func configureCell(viewModel: ArticleViewModel, indexPath: IndexPath) {
+    @IBAction func addToFavorites(_ sender: UIButton) {
+        guard let viewModel = viewModel, let index = index else { return }
+        viewModel.toggleFavoriteForArticle(index: index)
+        isFavorite.toggle()
+        guard let article = viewModel.getArticle(index: index) else {return}
+        loadImage(with: article.urlToImage)
+    }
+    
+    
+    func loadImage(with string: String?) {
+        guard let viewModel = viewModel, let index = index else { return }
+        guard let article = viewModel.getArticle(index: index) else {return}
+        guard let string else {
+            let defaultImage = UIImage(named: "defaultImage")
+            let defaultImageData = defaultImage!.pngData()
+            self.coreDataService.addArticle(article: article, imageData: defaultImageData!)
+            return
+        }
+        viewModel.downloadImage(urlString: string) { image in
+            if let image {
+                self.coreDataService.addArticle(article: article, imageData: image)
+            }
+        }
+    }
+    
```

Comment:
try to have only one empty line between methods. In the project we have SwiftLint which will throw a warning for that